lecture_3/03_project/calculator/internal/infrastructure/sqlite/betRepository.go

```diff
@@ -0,0 +1,186 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/pkg/errors"
+
+	domainmodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/domain/models"
+	storagemodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/sqlite/models"
+)
+
+// BetRepository provides methods that operate on cals_bets SQLite database.
+type BetRepository struct {
+	dbExecutor DatabaseExecutor
+	betMapper  BetMapper
+}
+
+// NewBetRepository creates and returns a new BetRepository.
+func NewBetRepository(dbExecutor DatabaseExecutor, betMapper BetMapper) *BetRepository {
+	return &BetRepository{
+		dbExecutor: dbExecutor,
+		betMapper:  betMapper,
+	}
+}
+
+// InsertBetCalculated inserts the provided calculated bet into the database. An error is returned if the operation
+// has failed.
+func (r *BetRepository) InsertBetCalculated(ctx context.Context, bet domainmodels.BetCalculated) error {
+	storageBet := r.betMapper.MapDomainBetToStorageBet(bet)
+	err := r.queryInsertBet(ctx, storageBet)
+	if err != nil {
+		return errors.Wrap(err, "calc_bets repository failed to insert a bet with id "+bet.Id)
+	}
+	return nil
+}
+
+func (r *BetRepository) queryInsertBet(ctx context.Context, bet storagemodels.BetCalculated) error {
+	// If payout is 0, do not insert it.
+	if bet.Payment == 0 {
```

Comment:
This is not needed. In Controller I wanted to check whether `payout` is 0 and not insert it into DB if it is. However, for `payment` we can assume it will never be 0 (because a bet with payment of 0 shouldn't even enter the system, that bet should be refused). You can safely insert in into the DB without having this check.